### PR TITLE
Dev/prv erc20

### PR DESCRIPTION
**Pull request by thachtb**

### CODE REVIEW

File: blockchain/bridgetxsbuilder.go [hunk 1]
```diff
@@ -315,6 +319,7 @@ func (blockGenerator *BlockGenerator) buildBridgeIssuanceTx(
 	shardView *ShardBestState,
 	featureStateDB *statedb.StateDB,
 	metatype int,
+	isPRVPegging bool,
```
Comment (dat-incognito-org) [inline blockchain/bridgetxsbuilder.go:322]: should be called `isPeggedPRV`. The widely adopted term is `pegged <TOKEN>`. Issuing a pegged token isn't quite the same as pegging that token.

File: metadata/burningrequest.go [hunk 2]
```diff
@@ -87,14 +87,26 @@ func (bReq BurningRequest) ValidateSanityData(chainRetriever ChainRetriever, sha
...
-	if shardViewRetriever.GetEpoch() < config.Param().ETHRemoveBridgeSigEpoch && (bReq.Type == BurningRequestMetaV2 || bReq.Type == BurningForDepositToSCRequestMetaV2 || bReq.Type == BurningPBSCRequestMeta) {
+	if shardViewRetriever.GetEpoch() < config.Param().ETHRemoveBridgeSigEpoch &&
+		(bReq.Type == BurningRequestMetaV2 || bReq.Type == BurningForDepositToSCRequestMetaV2 || 
+		bReq.Type == BurningPBSCRequestMeta || bReq.Type == BurningPRVERC20RequestMeta || 
+		bReq.Type == BurningPRVBEP20RequestMeta ) {
 		return false, false, fmt.Errorf("metadata type %d is not supported", bReq.Type)
 	}
+
+	if (bReq.Type == BurningPRVERC20RequestMeta || bReq.Type == BurningPRVBEP20RequestMeta) && bReq.TokenID.String() != common.PRVIDStr {
+		return false, false, fmt.Errorf("1 metadata type %d does not support for incTokenID %v", bReq.Type, bReq.TokenID.String())
```
Comment (dat-incognito-org) [inline metadata/burningrequest.go:98]: why prepend these errors with `"1"` & `"2"` ?

File: rpcserver/constants.go [hunk 3]
```diff
@@ -176,24 +176,29 @@ const (
...
+	getAllBridgeTokens                    = "getallbridgetokens"
+	getAllBridgeTokensByHeight            = "getallbridgetokensbyheight"
+	getETHHeaderByHash                    = "getethheaderbyhash"
+	getBridgeReqWithStatus                = "getbridgereqwithstatus"
+	createAndSendTxWithIssuingBSCReq      = "createandsendtxwithissuingbscreq"
+	createAndSendTxWithIssuingPRVERC20Req = "createandsendtxwithissuingprverc20req"
+	createAndSendTxWithIssuingPRVBEP20Req = "createandsendtxwithissuingprvbep20req"
+	createAndSendBurningBSCRequest        = "createandsendburningbscrequest"
+	checkBSCHashIssued                    = "checkbschashissued"
+	checkPRVPeggingHashIssued             = "checkprvpegginghashissued"
```
Comment (dat-incognito-org) [inline rpcserver/constants.go:199]: should be `...PeggedPRV...`

File: metadata/burningrequest.go (hunk 2 above)
Comment (thachtb) [reply metadata/burningrequest.go]: removed prepend

File: rpcserver/constants.go (hunk 3 above)
Comment (thachtb) [reply rpcserver/constants.go]: update method name checkprvpegginghashissued -> checkpeggedprvhashissued

**Review by dat-incognito-org — APPROVED** [review]

**Review by 0xankylosaurus — APPROVED** [review]

**Pull request merged**